File: src/pocketcloud/cloud/event/impl/player/PlayerSwitchServerEvent.php
<?php

namespace pocketcloud\cloud\event\impl\player;

use pocketcloud\cloud\player\CloudPlayer;
use pocketcloud\cloud\server\CloudServer;

class PlayerSwitchServerEvent extends PlayerEvent {

    public function __construct(
        CloudPlayer $player,
        private readonly ?CloudServer $oldServer,
        private readonly CloudServer $newServer
    ) {
        parent::__construct($player);
    }

    public function getOldServer(): ?CloudServer {
        return $this->oldServer;
    }

    public function getNewServer(): CloudServer {
        return $this->newServer;
    }
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/event/impl/player/PlayerSwitchServerEvent.java
package de.pocketcloud.cloud.event.impl.player;

import de.pocketcloud.cloud.player.CloudPlayer;
import de.pocketcloud.cloud.server.CloudServer;
import lombok.Getter;
import org.jetbrains.annotations.Nullable;

public final class PlayerSwitchServerEvent extends PlayerEvent {

    @Getter
    @Nullable
    private final CloudServer oldServer;
    @Getter
    private final CloudServer newServer;

    public PlayerSwitchServerEvent(CloudPlayer player, @Nullable CloudServer oldServer, CloudServer newServer) {
        super(player);
        this.oldServer = oldServer;
        this.newServer = newServer;
    }
}
